en/configure-a-tidb-cluster.md

```diff
@@ -1,35 +1,252 @@
+    replicas: 1
+    service:
+      type: ClusterIP
+    config:
+      split-table: true
+      oom-action: "log"
+    requests:
+      cpu: 1
+```
+
+For all the configurable parameters of TiDB, refer to [TiDB Configuration File](https://pingcap.com/docs/v3.1/reference/configuration/tidb-server/configuration-file/).
+
+> **Note:**
+>
+> If you deploy your TiDB cluster using CR, make sure that `Config: {}` is set, no matter you want to modify `config` or not. Otherwise, TiDB components might not be started successfully. This step is meant to be compatible with `Helm` deployment.
+
+### Configure TiKV parameters
+
+TiKV parameters can be configured by `spec.tikv.config` in TidbCluster Custom Resource.
+
+For example:
+
+```yaml
+apiVersion: pingcap.com/v1alpha1
+kind: TidbCluster
+metadata:
+  name: basic
+spec:
+....
+  tikv:
+    image: pingcap.com/tikv:v3.1.0
+    config:
+      grpc-concurrenc: 4
+      sync-log: true
+    replicas: 1
+    requests:
+      cpu: 2
+```
+
+For all the configurable parameters of TiKV, refer to [TiKV Configuration File](https://pingcap.com/docs/v3.1/reference/configuration/tikv-server/configuration-file/).
+
+> **Note:**
+>
+> If you deploy your TiDB cluster using CR, make sure that `Config: {}` is set, no matter you want to modify `config` or not. Otherwise, TiKV components might not be started successfully. This step is meant to be compatible with `Helm` deployment.
+
+### Configure PD parameters
+
+PD parameters can be configured by `spec.pd.config` in TidbCluster Custom Resource.
+
+For example:
+
+```yaml
+apiVersion: pingcap.com/v1alpha1
+kind: TidbCluster
+metadata:
+  name: basic
+spec:
+.....
+  pd:
+    image: pingcap.com/pd:v3.1.0
+    config:
+      format: "format"
+      disable-timestamp: false
+```
+
+For all the configurable parameters of PD, refer to [PD Configuration File](https://pingcap.com/docs/v3.1/reference/configuration/pd-server/configuration-file/).
+
+> **Note:**
+>
+> If you deploy your TiDB cluster using CR, make sure that `Config: {}` is set, no matter you want to modify `config` or not. Otherwise, PD components might not be started successfully. This step is meant to be compatible with `Helm` deployment.
+
+### Configure TiFlash parameters
+
+TiFlash parameters can be configured by `spec.tiflash.config` in TidbCluster Custom Resource.
+
+For example:
+
+```yaml
+apiVersion: pingcap.com/v1alpha1
+kind: TidbCluster
+metadata:
+  name: basic
+spec:
+  ...
+  tiflash:
+    config:
+      config:
+        logger:
+          count: 5
+          level: information
+```
+For all the configurable parameters of TiFlash, refer to [TiFlash Configuration File](https://pingcap.com/docs/v3.1/tiflash/tiflash-configuration/).
+
+> **Note:**
+>
+> If you deploy your TiDB cluster using CR, make sure that `Config: {}` is set, no matter you want to modify `config` or not. Otherwise, TiFlash components might not be started successfully. This step is meant to be compatible with `Helm` deployment.
```

Comment:
This can be removed, no need for TiFlash.

Reply:
Can I address this in a follow up PR?

Reply:
I forget to remove this.